odoo/api.py

```diff
@@ -638,7 +638,7 @@ def company(self):
         if company_ids:
             if not self.su:
                 user_company_ids = self.user._get_company_ids()
-                if any(cid not in user_company_ids for cid in company_ids):
+                if set(company_ids) - set(user_company_ids):
```

Comment:
Not sure that it makes sense to check for all companies instead of the only the first one (`company_ids[0]`).
(But it is a small change behavior).
-> `if company_ids[0] not in user_company_ids:`

Reply:
I kept exactly the same behavior, I think it is better in stable